setup.py

```diff
@@ -481,6 +481,7 @@ def run_setup():
         packages = [
             'mpi4py',
             'mpi4py.futures',
+            'mpi4py.util',
```

Comment:
No, this will not turn numpy in a hard test dependency. The tests are carefully coded to run without NumPy. Of course, a few tests are skipped, and others are not as meaningful and exhaustive given that the `numpy.dtype` -> `MPI.Datatype` conversion is not available.

Reply:
OK you are right about testing. But how about generating stubs? I noticed `src/mpi4py/util/dtlib.pyi` imports numpy, so looks like for the purpose of generating docs numpy is a dependency? (I'm totally fine with it, just curious.)

Reply:
AFAICT, documentation generation does not depend on *.pyi stub files, although perhaps recent Sphinx versions (or third-party extensions) look at them, I really do not know. Stub files are about type checking and not about doc generation. Those using type-checkers will need NumPy installed, of configure things to ignore a missing numpy. Remember that type-checking is "static", I cannot protect `import numpy` with try/except. 

FYI, the only *.pyi stubs file that is autogenerated in full is `mpi4py/MPI.pyi` using `conf/mpistubgen.py`. All the other stub fiels where either written by hand or autogenerated (with mypy's stubgen script) and heavily tweaked by hand afterwards.

Reply:
> FYI, the only *.pyi stubs file that is autogenerated in full is `mpi4py/MPI.pyi` using `conf/mpistubgen.py`. All the other stub fiels where either written by hand or autogenerated (with mypy's stubgen script) and heavily tweaked by hand afterwards.

Yeah I just realized it...Sorry for missing that until now. I feel so sad that you have to maintain all other `*.pyi` files by hand...Is it a necessary evil?

Reply:
Well, I believe NumPy folks are following the exact same approach. The hard part is to get the stubs the first time, afterwards maintenance is relatively easy. I still have to setup mypy's stubtest linter (and other linting steps like running pylint) on CI. I'll do it shortly after we merge this PR.

PS: Back in the days, when I was a junior, Python folks bragged about duck typing and that Python was awesome because you did not have to declare variables and its types. But the usual inability of humans of writing a couple lines of code without introducing a bug gave a dosis of reality and put the community in check.  Now the key players are advocating typing and type checkers and all that. However, there is a problem... aside from the compatibility issues (Python 2), the yet-evolving API, and the extra dependencies you may need to inject in your project (e.g. `typing_extensions`), the cruel fact is that Python typing is still an afterthought. If you do not want to maintain *.pyi files, then you have to add typing in your regular Python code. And then you open the door to tons of nasty typing-related code with exact zero-benefit at runtime, all you get is a sort of linter on steroids. Please go to NumPy sources and look at the *.pyi files. If you wanted to merge/combine all that mess on the original *.py sources, you can see how much noise you add to otherwise perfectly good looking Python code.